src/app/convai-checker.component.ts

```diff
@@ -229,6 +230,7 @@ export class ConvaiChecker implements OnInit, OnChanges {
       this.analyzeCommentResponse = null;
       this.analyzeCommentResponseChanged.emit(this.analyzeCommentResponse);
       this.statusWidget.notifyScoreChange(0);
+      this.scoreChanged.emit(0);
```

Comment:
These lines seem redundant from my first glance but that's probably a lack of familiarity.  What's the reason to have then notify callback on the child element instead of passing the value as a parameters?

Reply:
Do you mean instead of using data binding? This is because the data binding was out of sync. Basically, the score was (previously) being updated via data binding, but setLoading was called inline. So we could end up with a scenario where the http request completed between calls to Angular's change detection, and setLoading(false) was called, but the change detection hadn't run so that even though the response was stored in this (the parent) component, the child hadn't been notified yet. And both values couldn't be passed via data binding because that relied on using ngOnChanges to call the methods in the child, and for some reason I suspect is due to weirdness with gsap animation callbacks, ngOnChanges wasn't getting triggered while the animation was playing, which meant the animation would go on forever. So the only option left was to notify of the score change directly. The PR where I made this change is still open here: https://github.com/conversationai/perspectiveapi-authorship-demo/pull/16/files Might make more sense to look at it first to see where things were. 

Happy to chat about this in person also, and there's an internal thread on the angular mailing list where I asked about this.

Reply:
That makes sense, but seems a bit unfortunate.